src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/FailedToPrintTaskTest.java

```diff
@@ -0,0 +1,185 @@
+package uk.gov.hmcts.reform.sendletter.tasks;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.hmcts.reform.sendletter.entity.Letter;
+import uk.gov.hmcts.reform.sendletter.entity.LetterRepository;
+import uk.gov.hmcts.reform.sendletter.entity.LetterState;
+import uk.gov.hmcts.reform.sendletter.logging.AppInsights;
+import uk.gov.hmcts.reform.sendletter.services.FtpAvailabilityChecker;
+import uk.gov.hmcts.reform.sendletter.util.MessageIdProvider;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(SpringRunner.class)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest
+public class FailedToPrintTaskTest {
+
+    @Autowired
+    LetterRepository repository;
+
+    @Autowired
+    private FtpAvailabilityChecker availabilityChecker;
+
+    private LocalTime cutOff;
+
+    private LocalTime secondBeforeCutOff;
+
+    @Mock
+    private AppInsights insights;
+
+    private FailedToPrintTask task;
+
+    @Before
+    public void setUp() {
+        task = new FailedToPrintTask(repository, insights, availabilityChecker);
+        cutOff = availabilityChecker.getDowntimeStart();
+        secondBeforeCutOff = cutOff.minusSeconds(1);
+
+        // making sure each test has fresh table
+        repository.deleteAll();
+    }
+
+    @Test
+    public void should_return_204_when_there_are_no_unprinted_letters() {
+        // given
+        ArgumentCaptor<Letter> captor = ArgumentCaptor.forClass(Letter.class);
+
+        // when
+        task.run();
+
+        // then
+        verify(insights, never()).trackNotPrintedLetter(captor.capture());
+
+        // and
+        assertThat(captor.getAllValues()).isEmpty();
+    }
+
+    @Test
+    @SuppressWarnings("VariableDeclarationUsageDistance")
+    public void should_return_204_when_there_is_an_unprinted_letter() {
```

Comment:
?

Reply:
i'm guessing question mark is for warning suppression. checkstyle doesn't like my bdd given/when/then setup - complains letter is too far from usage